### ISSUE TITLE
[Accessibility][Product Teaser] Both the product image and product name are linked to the product page

### ISSUE THREAD

**Issue by LevelAccessA11y** [adobe/aem-core-wcm-components#1769]
**Steps to Reproduce**
1. Enable NVDA screen reader.
2. Open page in Google Chrome.
3. Use TAB and/or SHIFT + TAB to navigate through the product teaser.

**Current Behavior**
The link text and product image are separate links serving the same purpose, which can make for a cumbersome experience as screen reader users navigate through the product teaser element.

**Expected behavior/code**
When links serve the same purpose, they should be identified by the same accessible name. In this case, because the image and link are immediate siblings in the focus order, it makes sense to only link one to reduce the navigation needed for keyboard-only users. To capture image clicks and link clicks, use Javascript.

**Environment**
Chrome 91, Windows 10, NVDA 2020.4 
AEM Core Components v. 2.17  

**Affected Users**
Keyboard-only users, screen reader users

**WCAG SC**
2.4.4 Link Purpose (In Context)

**Comment by richardhand** [comment]
@adobe export issue to Jira project SITES

**Comment by mhaack** [comment]
Tracked via CIF-2366
